orchagent/orch.cpp

```diff
@@ -370,6 +371,60 @@ ref_resolve_status Orch::resolveFieldRefValue(
     return ref_resolve_status::success;
 }
 
+void Orch::setObjectReference(
+    type_map &type_maps,
+    const string &table,
+    const string &obj_name,
+    const string &field,
+    const string &referenced_table,
+    const string &referenced_obj)
+{
+    auto &obj = (*type_maps[table])[obj_name];
+    auto field_ref = obj.m_objsReferencingByMe.find(field);
+
+    // remove the reference to the old object, if any
+    if (field_ref != obj.m_objsReferencingByMe.end())
+    {
+        auto &old_referenced_obj_name = field_ref->second;
+        vector<string> tokens = tokenize(old_referenced_obj_name, list_item_delimiter);
+        for (auto &token : tokens)
+        {
+            // obj_name references token
+            auto &old_referenced_obj = (*type_maps[referenced_table])[token];
+            old_referenced_obj.m_objsDependingOnMe.erase(obj_name);
+            SWSS_LOG_INFO("Obj %s Field %s: Remove reference to %s", obj_name.c_str(), field.c_str(), token.c_str());
```

Comment:
Shall we set the log level to debug? and also apply for below logs in this function.

Reply:
I don't prefer to use DEBUG. looks like DEBUG level is used when the info is trivial but these ones are relatively important.